### DISCUSSION TITLE
Aspect oriented programming with aliases.

### DISCUSSION THREAD

**Discussion by WalterWoshid** [laravel/framework#39952] [Ideas]
Following situation:

You've got a controller like this:
```php
class IndexController extends Controller
{
    public function index()
    {
        return 'Important value';
    }
}
```

and you'd like to create an alias in the service provider to override this class like that:
```php
class AppServiceProvider extends ServiceProvider
{
    public function register()
    {
        AliasLoader::getInstance()->alias(IndexController::class, DifferentController::class);
    }
}

class DifferentController extends Controller
{
    public function index()
    {
        return 'Important value and more important values';
    }
}
```

But what if you want to keep most of the functionality of the `IndexController` from the `DifferentController` like this:
```php
class DifferentController extends IndexController
{
    public function index()
    {
        return parent::index() . ' and more important values';
    }
}
```

Sadly the code above does not work, because apparently it's a circular logic, but wouldn't it be awesome to have such a functionality? This would technically allow aspect oriented programming. Please correct me if I'm wrong.

**Comment by mikield** [comment]
I think that this I how it should work, considering if Controller is made from a container.
